statsmodels/stats/multicollinearity.py

```diff
@@ -0,0 +1,460 @@
+        Optional moment or correlation matrix to be used for the analysis. If it is provided
+        then data is ignored.
+        The moment_matrix needs to be nonsingular because Cholesky decomposition is used.
+    standardize : bool, default True
+        If true and data are provided, then the data will be standardized to mean zero and
+        standard deviation equal to one, which is equivalent to using the correlation matrix.
+        TODO: This might be replaced by separate demean option. Variance inflation factor
+        can be calculated for data that is scaled but not demeaned.
+        See Notes in class MultiCollinear
+
+    Return
+    ------
+    vif : ndarray or pandas.Series
+        variance inflation factor. This will be a pandas Series if the data has a `columns`
+        attribute as provided by a pandas DataFrame.
+
+    """
+    # most parts are duplicate code, copied from vif_selection
+    x = np.asarray(data)
+    # TODO: use pandas corrcoef below to have nan handling ?
+
+    if moment_matrix is not None:
+        xm = np.asarray(moment_matrix)
+        if standardize:
+            # check if we have correlation matrix,
+            # we cannot demean but we can scale
+            # (We could demean if const is included in uncentred moment matrix)
+            if (np.diag(xm) != 1).any():
+                xstd = np.sqrt(np.diag(xm))
+                xm /= np.outer(xstd, xstd)
+    else:
+        if standardize:
+            xm = np.corrcoef(x, rowvar=0)
+            if np.any(np.ptp(x, axis=0) == 0):
+                # TODO: change this? detect from xcorr, drop constant ?
+                raise ValueError('If standardize is true, then data should ' +
+                                 'not include a constant')
+        else:
+            xm = np.dot(x.T, x) / float(x.shape[0])
+
+    vif_ = np.abs(np.diag(np.linalg.inv(xm)))
+
+    if hasattr(data, 'columns'):
+        import pandas
+        return pandas.Series(vif_, index=data.columns)
+    else:
+        return vif_
+
+
+def vif_selection(data, threshold=10, standardize=True, moment_matrix=None):
+    """Recursively drop variables if VIF is above threshold
+
+    default threshold is just a number, maybe don't use a default
+
+    Notes
+    -----
+    This avoids a loop to calculate the vif at a given set of variables, but
+    has one matrix inverse at each set.
+
+    TODO: replace inv by sweep algorithm,
+    question: how do we start sweep in backwards selection?
+
+
+    Note: This function tries to remove the last variable in case of ties in
+    vif. However, which variable of those that have theoretically equal vif
+    may be random because of floating point imprecision.
+
+    Parameters
+    ----------
+    data : array_like
+        data with observations in rows and variables in columns. This is ignored if
+        moment_matrix is not None. Data can be of smaller than full rank.
+    moment_matrix : None or ndarray
```

Comment:
threshold is not documented

sequence of moment_matrix and standardize is reversed, also in other docstrings